source/javascripts/lib/wirsing/filters/switch.coffee

```diff
@@ -0,0 +1,17 @@
+@wirsing ?= {}
+wirsing.filter ?= {}
+
+wirsing.filter.switch =
+  on: (items, key) ->
+    switchObject = {}
+
+    switches = ->
+      if Object.keys(switchObject).length == 0
+        switchObject[item[key]] ?= true for item in items
+      switchObject
+
+    names: -> Object.keys switches()
```

Comment:
It's actually at the end of the function, but the blank lines in between make it confusing :)

But I agree that your suggestion is much more intuitive.